source/extensions/filters/http/gzip/gzip_filter.h

```diff
@@ -62,9 +94,10 @@ class GzipFilterConfig {
   int32_t window_bits_;
 
   StringUtil::CaseUnorderedSet content_type_values_;
-
   bool disable_on_etag_header_;
   bool remove_accept_encoding_header_;
+  const std::string stats_prefix_;
+  Stats::Scope& scope_;
```

Comment:
Instead of storing the scope here, you actually want to create the stats object in the config instead of the filter, and the return that on request. See other similar examples in other filters.

Reply:
oh got it. didn't see this until now.